modules/manager/controller_setup.go

```
@@ -378,6 +381,20 @@ func SetupControllers(mgr manager.Manager, c *Config) (map[string]ControllerDef,
 		}
 	}
 
+	clusterCAKeyConfig := secrets.KeyConfig{
+		Type: func() x509.PublicKeyAlgorithm {
+			switch c.ClusterCAKeyType {
+			case mgrconfig.RSA:
+				return x509.RSA
+			case mgrconfig.ECDSA:
+				return x509.ECDSA
+			default:
+				return x509.UnknownPublicKeyAlgorithm
+			}
```

This would be a fatal error, shouldn't we abort setup in such a case?

---

This would be caught at the CLI flag parsing level but you're right, we should also error out here when we detect an unsupported type

PTAL 7365feb61cab4cb5e90fe54dcae909c3ad26dc78